### ISSUE TITLE
withAxiomRouteHandler does not pass next route context

### ISSUE THREAD

**Issue by amrdraz** [axiomhq/next-axiom#149]
when using `^1.0.0-rc.1`

For a route like GET `api/posts/[id]`

The route Handler does not pass the params that come form the context as expected in the [docs](https://nextjs.org/docs/app/api-reference/file-conventions/route#parameters)

```ts
export const GET = withAxiomRouteHandler((request: AxiomRequest, { params }) => { ... })
```
where params would be in this case of type `{ params: { id: string } }` based on the path

I looked at the [line](https://github.com/axiomhq/next-axiom/blob/main/src/withAxiom.ts#L59)
and it looks like you need to do this

``ts
 return async (req: Request | NextRequest, ctx) => { // <- add ctx
    let region = '';
    if ('geo' in req) {
      region = req.geo?.region ?? '';
    }

    const report: RequestReport = {
      startTime: new Date().getTime(),
      path: req.url,
      method: req.method,
      host: req.headers.get('host'),
      userAgent: req.headers.get('user-agent'),
      scheme: 'https',
      ip: req.headers.get('x-forwarded-for'),
      region,
    };
    const isEdgeRuntime = globalThis.EdgeRuntime ? true : false;

    const logger = new Logger({ req: report, source: isEdgeRuntime ? 'edge' : 'lambda' });
    const axiomContext = req as AxiomRequest;
    axiomContext.log = logger;

    try {
      const result = await handler(axiomContext, ctx); // <- pass ctx
```

**Comment by amrdraz** [comment]
this is my current patch file


```diff
diff --git a/node_modules/next-axiom/dist/withAxiom.d.ts b/node_modules/next-axiom/dist/withAxiom.d.ts
index 2f93fd6..996420c 100644
--- a/node_modules/next-axiom/dist/withAxiom.d.ts
+++ b/node_modules/next-axiom/dist/withAxiom.d.ts
@@ -5,11 +5,11 @@ declare global {
     var EdgeRuntime: string;
 }
 export declare function withAxiomNextConfig(nextConfig: NextConfig): NextConfig;
-export type AxiomRequest = (NextRequest | Request) & {
+export type AxiomRequest = (NextRequest) & {
     log: Logger;
 };
-type NextHandler = (req: AxiomRequest) => Promise<Response> | Promise<NextResponse> | NextResponse | Response;
-export declare function withAxiomRouteHandler(handler: NextHandler): (req: Request | NextRequest) => Promise<Response | NextResponse<unknown>>;
+type NextHandler = (req: AxiomRequest, context?: { params: any }) => Promise<Response> | Promise<NextResponse> | NextResponse | Response;
+export declare function withAxiomRouteHandler(handler: NextHandler): (req: NextRequest) => Promise<Response | NextResponse<unknown>>;
 export declare function withAxiom(param: NextConfig): NextConfig;
 export declare function withAxiom(param: NextHandler): NextHandler;
 export {};
diff --git a/node_modules/next-axiom/dist/withAxiom.js b/node_modules/next-axiom/dist/withAxiom.js
index 104eb53..4f26277 100644
--- a/node_modules/next-axiom/dist/withAxiom.js
+++ b/node_modules/next-axiom/dist/withAxiom.js
@@ -52,7 +52,7 @@ function withAxiomNextConfig(nextConfig) {
 }
 exports.withAxiomNextConfig = withAxiomNextConfig;
 function withAxiomRouteHandler(handler) {
-    return (req) => __awaiter(this, void 0, void 0, function* () {
+    return (req, ctx) => __awaiter(this, void 0, void 0, function* () {
         var _a, _b;
         let region = '';
         if ('geo' in req) {
@@ -73,7 +73,7 @@ function withAxiomRouteHandler(handler) {
         const axiomContext = req;
         axiomContext.log = logger;
         try {
-            const result = yield handler(axiomContext);
+            const result = yield handler(axiomContext, ctx);
             yield logger.flush();
             if (isEdgeRuntime) {
                 logEdgeReport(report);

```

**Comment by schehata** [comment]
Hi @amrdraz, thanks for your contribution. We have an [open PR](https://github.com/axiomhq/next-axiom/pull/148) to fix the issue, I will review that against your patch. Once we merge this fix, we will release v1.0.0, I expect to be able to do that today.

**Comment by schehata** [comment]
I've found one error when doing a production build, in our examples dir:

```
Type error: Route "app/api/dynamic/[id]/route.ts" does not match the required types of a Next.js Route.
  Invalid configuration "GET":
```

this seems to be a conflict in `withAxiom` type, for some reason when params is passed, typescript thinks that this is withAxiom for NextConfig, and thus fails the conditions for being a route handler.

**Comment by schehata** [comment]
so if I use 

```
type RouteHandler = (
  req: AxiomRequest,
  ctx?: any
) => Promise<Response> | Promise<NextResponse> | NextResponse | Response;
```

that works fine with `withAxiomRouteHandler` but fails with `withAxiom`. it still returns `NextConfig` for some reason.

**Comment by rajdtta** [comment]
Just chiming in because I'm also encountering this issue.

@schehata, do you think that `withAxiom` returns `NextConfig` because of this function? (located in `/src/withAxiom.ts`)

```
function isNextConfig(param: WithAxiomParam): param is NextConfig {
  return typeof param == 'object';
}
```

Only skimmed through the codebase, but that function (and its usage as part of the `withAxiom()` export) stands out to me.

**Comment by schehata** [comment]
are you still facing this issue? I remember it was fixed but can't remember if it was released, but probably it was in v1.1.0

**Comment by amrdraz** [comment]
It was fixed
